core/controllers/admin.py

```diff
@@ -656,6 +658,453 @@ def _generate_dummy_explorations(
+            should_submit_suggestions: boolean.
+                Track whether suggestions should be submitted.
+            story_url_fragment: stringType. Story fragment of existing story.
+        """
+        story = story_fetchers.get_story_by_url_fragment(story_url_fragment)
+        story_id = story.id
+        len_nodes = len(story.story_contents.nodes)
+
+        topic = topic_fetchers.get_topic_by_url_fragment('sample-topic-one')
+        topic_id = topic.id
+
+        story_node_dicts = self._generate_sample_chapters(
+            num_sample_ops, num_sample_interactions_per_opportunity, len_nodes)
+
+        story_change_list = []
+        len_story = len(story_node_dicts)
+        for i, story_node_dict in enumerate(story_node_dicts, 1):
+            self._generate_sample_story_node(
+                story, len_story, len_nodes + i, **story_node_dict)
+            cur_node_id = 'node_' + str(len_nodes + i)
+            story_change_list.append(story_domain.StoryChange({
+                'cmd': 'add_story_node',
+                'node_id': cur_node_id,
+                'title': story_node_dict['title']
+            }))
+            story_change_list.append(story_domain.StoryChange({
+                'cmd': 'update_story_node_property',
+                'property_name': 'exploration_id',
+                'old_value': None,
+                'new_value': story_node_dict['exp_id'],
+                'node_id': cur_node_id,
+            }))
+            story_change_list.append(story_domain.StoryChange({
+                'cmd': 'update_story_node_property',
+                'property_name': 'description',
+                'old_value': None,
+                'new_value': story_node_dict['description'],
+                'node_id': cur_node_id,
+            }))
+
+        topic_services.update_story_and_topic_summary(
+            self.user_id,
+            story_id,
+            story_change_list,
+            'adding more opportunities',
+            topic_id)
+
+        exp_ids_in_story = story.story_contents.get_all_linked_exp_ids()[
+            len_nodes:]
+        opportunity_services.add_new_exploration_opportunities(
+            story_id, exp_ids_in_story)
+
+        if should_submit_suggestions:
+            self._submit_suggestions(
+                num_sample_ops,
+                num_sample_interactions_per_opportunity,
+                exp_ids_in_story)
+
+    def _submit_suggestions(
+        self,
+        num_sample_ops,
+        num_sample_interactions_per_opportunity,
+        exp_ids_in_story):
+        """Submits suggestions to explorations linked to the given exp_ids.
+
+        Args:
+            num_sample_ops: int. Number of sample opportunities that
+                have been generated.
+            num_sample_interactions_per_opportunity: int. Number of sample
+                interactions that have been generated.
+            exp_ids_in_story: list. List of the exploration ids of explorations
```

Comment:
Nit: Since we specify the "list" type already, we can just omit the "List of" part of the description. Here and below.